File: python/nivel 4 trampas/24.py
"""
Ejercicio 24
Determinar si dos palabras son anagramas
"""
palabra1 = input("Ingresa la primera palabra: ").lower()
palabra2 = input("Ingresa la segunda palabra: ").lower()

if len(palabra1) != len(palabra2):
  print("No son anagramas")
else:
  conteo1 = {}
  conteo2 = {}

  for letra in palabra1:
    conteo1[letra] = conteo1.get(letra, 0) +1

  for letra in palabra2:
    conteo2[letra] = conteo2.get(letra, 0) +1

  if conteo1 == conteo2:
    print("las palabras son anagramas")
  else:
    print("las palabras NO son anagramas")

File: javascript/nivel 4 trampas/24.js
/*
Ejercicio 24
Determinar si dos palabras son anagramas
*/
let palabra1 = prompt("Ingresa la primera palabra: ").toLowerCase();
let palabra2 = prompt("Ingresa la segunda palabra: ").toLowerCase();

if(palabra1.length !== palabra2.length){
  console.log("no son anagramas")
}else{
  let conteo = {};

  //contar letras de la primera palabra
  for(let letra of palabra1){
    if(conteo[letra]){
      conteo[letra]++;
    }else{
      conteo[letra] = 1;
    }
  }

  //restar letras usando la segunda palabra
  let esAnagrama = true;

  for(let letra of palabra2){
    if(!conteo[letra]){
      conteo[letra]++
    }else{
      conteo[letra] --;
    }
  }

  if(esAnagrama){
    console.log("Las palabras SON Anagramas");
  }else{
    console.log("Las palabras NO SON Anagramas");
  }




}